server/src/main/java/org/apache/cassandra/sidecar/modules/guice-best-practice.md

```diff
@@ -0,0 +1,117 @@
+<!--
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+-->
+
+# Guice Best Practices in Sidecar
+
+First of all, refer to [Guice Wiki](https://github.com/google/guice/wiki/BestPractices) for the best practices in general.
+
+There are certain project-specific best practices we advocate for Apache Sidecar. This article capture those items in the below. 
+
+## Multi-bindings for Plugin Extension
+
+[Multi-bindings](https://github.com/google/guice/wiki/Multibindings) is a Guice feature to support the plugin-type architecture,
+where multiple modules contribute components to plug into the system.
+
+The applications of multi-bindings in Sidecar are, for example,
+1. Routes definition
+2. Periodic tasks deployment
+3. Sidecar internal schemas creation
```

Comment:
I don't think this is scoped to sidecar internal schemas only, for example system_auth is also managed by multi-bindings, as a read-only table

Reply:
I updated the list. Please take another look if it sounds better.